rule/rule.go

```diff
@@ -730,8 +730,16 @@ type attrValue struct {
 
 // NewRule creates a new, empty rule with the given kind and name.
 func NewRule(kind, name string) *Rule {
+	var call *bzl.CallExpr
 	kindIdent := &bzl.Ident{Name: kind}
-	call := &bzl.CallExpr{X: kindIdent}
+
+	parts := strings.Split(kind, ".")
+	if len(parts) > 1 {
+		call = &bzl.CallExpr{X: createDotExpr(parts)}
+	} else {
+		call = &bzl.CallExpr{X: kindIdent}
+	}
```

Comment:
Could you extract out this part into `createDotExpr`? It can return a generic `Expr` that is either a `DotExpr` or an `Ident`.

Reply:
Sure, changes updated.